cmd/minishift/cmd/image/export.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package image
+
+import (
+	"fmt"
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/docker/image"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var imageExportCmd = &cobra.Command{
+	Use:   "export [image ...]",
+	Short: "Exports container images (experimental).",
+	Long:  "Exports container images (experimental).",
+	Run:   exportImage,
+}
+
+func exportImage(cmd *cobra.Command, args []string) {
+	logFile := createLogFile()
+	defer logFile.Close()
+
+	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+	defer api.Close()
+
+	if len(args) < 1 {
+		atexit.ExitWithMessage(0, "No image specifications provided")
+	}
+
+	util.ExitIfUndefined(api, constants.MachineName)
+
+	host, err := api.Load(constants.MachineName)
+	if err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error creating VM client: %v", err))
```

Comment:
_err_ is not always used, so it would need to be optional. Also the exit code would need to be passed. I think we are not leveraging the fact that we can return more than 0 or 1 yet (but we might).

Also I find the current ExitwithMessage a bit more self explanatory. ExitWithError leaves some questions open and one might have to look into the impl more often to see what's going on.

That said, I am not against a refactoring, but in this case I think it should be across the code base and in a separate issue. Right now we finally made it to be consistent in the use of ExitWithMessage. If introducing something new, it should be across the board.